src/rdb_protocol/terms/string.cc

```diff
@@ -125,6 +127,66 @@ class match_term_t : public op_term_t {
     virtual const char *name() const { return "match"; }
 };
 
+class format_term_t : public op_term_t {
+public:
+    format_term_t(compile_env_t *env, const raw_term_t &term)
+            : op_term_t{env, term, argspec_t{2}}, compile_env{env} {}
+private:
+    virtual scoped_ptr_t <val_t> eval_impl(scope_env_t *env, args_t *args, eval_flags_t) const {
+        const std::string templ = args->arg(env, 0)->as_str().to_std();
+        const datum_t datum = args->arg(env, 1)->as_datum();
+
+        const size_t template_length = templ.size();
+        std::string formatted{};
+
+        for (size_t pos = 0; pos < template_length; ++pos) {
+            rcheck(templ[pos] != '}',
+                   base_exc_t::LOGIC,
+                   "No parameter tags to close");
+
+            if (templ[pos] == '{') {
+                size_t param_end_pos {1};
+                std::string param_name {};
+
+                while (pos + param_end_pos < template_length && templ[pos + param_end_pos] != '}') {
+                    char next_char{templ[pos + param_end_pos]};
+
+                    rcheck(next_char != '{',
+                           base_exc_t::LOGIC,
+                           "Nested template parameters are not allowed");
+
+                    param_name.push_back(next_char);
+                    ++param_end_pos;
+                }
+
+                rcheck(pos + param_end_pos != template_length,
+                       base_exc_t::LOGIC,
+                       "Parameter tag must be closed");
+
+                const datum_t &field{datum.get_field(datum_string_t{param_name})};
+
+                minidriver_t r{backtrace()};
+                counted_t<const term_t> term = compile_term(
+                        compile_env,
+                        r.expr(field).coerce_to("STRING").root_term());
+
+                const datum_string_t &result{term->eval(env)->as_str()};
```

Comment:
This should be not a const reference.  The existing code is not broken, but making it a const reference offers no advantage, because the return value of as_str() is returned by value and will be copy-elided or at worst move-constructed into the local.

Using a const reference for the variable type has a downside, too: it looks like there *might* be a bug.  What if eval(env) returned a temporary but `as_str()` returned a const reference pointing into its parameter?  That's generally plausible.  In that case, the temporary would get destructed and `result` would be pointing into freed memory.  If the return type of `as_str()` was changed in the future, then there would be a bug.  So this line increases the rate of bugs by some small number.

Generally, you don't want to have local variables be const references.  Unless there is something specific you want to accomplish.

```suggestion
                const datum_string_t result{term->eval(env)->as_str()};
```